### PR TITLE
refine encode segment id map buffer

### PR DESCRIPTION
**Pull request by XinfengZhang**
attention: these change will break backward compatibility, the pre-assumption is no one is using them besides intel media driver and intel oneVPL runtime, if there are other customer,  it should be risky 
so, please help to check whether there are somebody are using segment id map buffer for avc or vp9 encoding.

### CODE REVIEW

File: va/va_enc_av1.h [hunk 1]
```diff
@@ -426,19 +428,19 @@ typedef struct _VAEncSegParamAV1 {
...
-     * in this map contains the segment id of a particular block.
-     * The element is indexed by raster scan order.
-     * The value of each entry should be in the range [0..7], inclusive.
-     */
-    uint8_t    *pSegmentMap;
-} VAEncSegMapBufferAV1;
+/**
+ * \brief AV1 Block Segmentation ID Buffer
+ *
+ * The application provides a buffer of VAEncMacroblockMapBufferType containing
```
Comment (rosetta-jpn) [inline va/va_enc_av1.h:435]: Do we need to set size and num_elements in vaCreateBuffer to width and height as well as in vp9, respectively?

File: va/va.h [hunk 2]
```diff
@@ -1474,6 +1482,17 @@ typedef union _VAConfigAttribValEncPerBlockControl {
 /** \brief Driver supports decode processing rate report  */
 #define VA_PROCESSING_RATE_DECODE                     0x00000002
 /**@}*/
+
+/** @name segment ID map block size */
+/**@{*/
```
[the remark this answers is not in the document]
Comment (XinfengZhang) [reply va/va.h:1487]: if we add them, to keep some basic compatible , the value should be 3, 4,  it looks a little strange, I suspect who are using them.

File: va/va_enc_av1.h (hunk 1 above)
Comment (XinfengZhang) [reply va/va_enc_av1.h:435]: it is a kind of backward combability issue, I want to unify them to a linear buffer. so, the question is: is there other user beside intel media driver , is there any application is using the older interface?

Comment (leyu-yao) [reply va/va_enc_av1.h:435]: Would it be a better solution if we keep the original behavior when calling vaCreateBuffer with VAEncMacroblockMapBufferType, and we define a new BufferType (e.g. named VAEncUnifiedSegmentationMapBufferType). 
There are some benefits but libva will a little bit complicated.
We can keep all the backward compability and suggest/guide app-developers to switch to the new/unified interface.

Comment (XinfengZhang) [reply va/va_enc_av1.h:435]: so, we support both for VP9?

Comment (leyu-yao) [reply va/va_enc_av1.h:435]: Yes, I would prefer that. 
It would benefit the application maintenance effort. 
However, it will make libva more complicated......

File: va/va.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (rosetta-jpn) [reply va/va.h:1487]: But segmentation is 8x8 in vp9 and 4x4 in av1. So a driver should be able to support up the size?

File: va/va_enc_av1.h [hunk 3]
```diff
@@ -148,8 +148,10 @@ typedef union _VAConfigAttribValEncAV1Ext1 {
          */
         uint32_t interpolation_filter          : 5;
         /**
-         * \brief Min segmentId block size accepted.
+         * \brief segmentId block size accepted.
          * Application need to send seg_id_block_size in PPS equal or larger than this value.
+         * one bit represent one block size defined as VA_SEGID_BLOCKXXXX
+         * should be (1 << VA_SEGID_BLOCKXXX | 1 << VA_SEGID_BLOCKXXX ... )
          */
         uint32_t min_segid_block_size_accepted : 8;
```
Comment (rosetta-jpn) [inline va/va_enc_av1.h:156]: Can we remove min_ from the variable name?

File: va/va_enc_av1.h [hunk 4]
```diff
@@ -670,7 +672,6 @@ typedef struct  _VAEncPictureParameterBufferAV1 {
      *  0: 16x16 block size, default value;
      *  1: 32x32 block size;
      *  2: 64x64 block size;
-     *  3: 8x8 block size.
```
Comment (rosetta-jpn) [inline va/va_enc_av1.h:673]: Why is this removed? 
For the consisency to vp9, should this be VA_SEGID_BLOCKXXX?

File: va/va.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (rosetta-jpn) [reply va/va.h:1501]: Ping. How do you think about this comment?

**Review by rosetta-jpn — CHANGES_REQUESTED** [review]

File: va/va.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (XinfengZhang) [reply va/va.h:1501]: I see , but we could add it later, at least , there are no such support from HW vendors?

File: va/va_enc_av1.h (hunk 3 above)
Comment (XinfengZhang) [reply va/va_enc_av1.h:156]: change the name will cause API compatible issue.

Comment (rosetta-jpn) [reply va/va_enc_av1.h:156]: Ack.

File: va/va_enc_av1.h (hunk 4 above)
Comment (XinfengZhang) [reply va/va_enc_av1.h:673]: updated.

**Comment by XinfengZhang** [comment]
@rosetta-jpn , updated. how about this version?

File: va/va_enc_vp9.h [hunk 5]
```diff
@@ -587,13 +595,41 @@ typedef struct _VAEncMiscParameterTypeVP9PerSegmantParam {
  * \brief VP9 Block Segmentation ID Buffer
  *
  * The application provides a buffer of VAEncMacroblockMapBufferType containing
- * the initial segmentation id for each 8x8 block, one byte each, in raster scan order.
- * Rate control may reassign it.  For example, a 640x480 video, the buffer has 4800 entries.
+ * the initial segmentation id for each block, block size is specified by seg_id_block_size
+ * in VAEncPictureParameterBufferVP9  , one byte each, in raster scan order.
+ * Rate control may reassign it.  For example, a 640x480 video, set_id_block_size is
```
Comment (leyu-yao) [inline va/va_enc_vp9.h:600]: There is a typo...  set_id_block_size

File: va/va_enc_vp9.h [hunk 6]
```diff
@@ -587,13 +595,41 @@ typedef struct _VAEncMiscParameterTypeVP9PerSegmantParam {
...
+/** \brief Attribute value for VAConfigAttribEncVP9. */
+typedef union _VAConfigAttribValEncVP9 {
+    struct {
+        /**
+         * \brief segmentId block size accepted.
+         * This is the granularity of segmentation map.
+         * one bit represent one block size defined as VA_SEGID_BLOCKXXXX
+         * should be (1 << VA_SEGID_BLOCKXXX | 1 << VA_SEGID_BLOCKXXX ... )
+         */
+        uint32_t segid_block_size_accepted :    8;
```
Comment (leyu-yao) [inline va/va_enc_vp9.h:616]: Av1 is using "seg_id_block_size", should we keep it consistent?

**Review by leyu-yao — APPROVED** [review]

File: va/va.h [hunk 7]
```diff
@@ -1491,6 +1491,19 @@ typedef union _VAConfigAttribValEncPerBlockControl {
...
+/** \brief each segmentID represent a 16x16 block */
+#define VA_SEGID_BLOCK_16X16                          0
+/** \brief each segmentID represent a 32x32 block */
+#define VA_SEGID_BLOCK_32X32                          1
+/** \brief each segmentID represent a 64x64 block */
+#define VA_SEGID_BLOCK_64X64                          2
+/** \brief each segmentID represent a 8x8 block */
+#define VA_SEGID_BLOCK_8X8                            3
```
Comment (rosetta-jpn) [inline va/va.h:1512]: nit: Shall we add VA_SEGID_BLOCK_4x4? AV1 is 4x4 seg map.

**Review by rosetta-jpn — APPROVED** [review]

**Comment by rosetta-jpn** [comment]
Sorry for so delayed review.
LGTM % nit. Feel free to submit the patches.
